src/hotspot/share/utilities/resourceHash.hpp

```diff
@@ -197,5 +212,39 @@ class ResourceHashtable : public ResourceObj {
   }
 };
 
+template<unsigned TABLE_SIZE, typename K, typename V>
+class FixedResourceHashtableStorage : public ResourceObj {
+  using Node = ResourceHashtableNode<K, V>;
+
+  Node* _table[TABLE_SIZE];
+protected:
+  FixedResourceHashtableStorage() {
+    memset(_table, 0, TABLE_SIZE * sizeof(Node*));
+  }
+
```

Comment:
Destructor should be non-public to prevent slicing.  Also need to consider what to do about copying.

Reply:
This class doesn't have a destructor. Should I add one and make it protected?

Reply:
I would.  Make it `= default`.